src/firefly/js/ui/panel/CollapsiblePanel.jsx

```diff
@@ -157,11 +162,13 @@ CollapsibleItemView.propTypes = {
     componentKey: string,
     isOpen: bool,
     header: oneOfType([string, func, element]),
+    expandedHeader: oneOfType([string, func, element]),
```

Comment:
Instead of adding new prop and slotProps, I suggest fixing the `header` prop so that it does what it was designed to do.  `header` should be:

```js
    header: oneOfType([node, func]),      // func is called with isOpen as a parameter
```